src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -0,0 +1,207 @@
+    case _                             => false
+  }
+
+  override def toString: String = s"NonEmpty$toSet"
+}
+
+object NonEmptySet {
+  private def apply[A](elem: A, others: Set[A]): NonEmptySet[A] = new NonEmptySet(others + elem)
+
+  /** Creates a `NonEmptySet` with the specified elements.
+   *  @tparam A      the type of the `NonEmptySet`'s elements
+   *  @param elem    an element of the created `NonEmptySet`
+   *  @param others  the remaining elements of the created `NonEmptySet`
+   *  @return a new `NonEmptySet` with elements `elem` and `others`
+   */
+  def apply[A](elem: A, others: A*): NonEmptySet[A] = apply(elem, others.toSet)
+
+  def unapply[A](arg: NonEmptySet[A]): Some[(A, Set[A])] = Some(arg.destruct)
+
+  /**
+   * Constructs a `NonEmptySet` from an element and `Set`.
+   */
+  def fromSet[A](elem: A, others: Set[A]): NonEmptySet[A] = apply(elem, others)
+
+  /**
+   * Constructs a `NonEmptySet` from a `Set` or `None` otherwise.
+   */
+  def fromSetOption[A](set: Set[A]): Option[NonEmptySet[A]] = set.headOption.map(fromSet(_, set.tail))
+
+  /**
+   * Constructs a `NonEmptySet` from an element and `Iterable`.
+   */
+  def fromIterable[A](head: A, tail: Iterable[A]): NonEmptySet[A] =
+    fromSet(head, tail.toSet)
+
+  /**
+   * Constructs a `NonEmptySet` from an `Iterable` or `None` otherwise.
+   */
+  def fromIterableOption[A](iterable: Iterable[A]): Option[NonEmptySet[A]] =
+    iterable.headOption.fold(None: Option[NonEmptySet[A]])(h => Some(fromIterable(h, iterable.tail)))
+
+  /**
+   * Constructs a `NonEmptySet` with the specified single value.
+   */
+  def single[A](head: A): NonEmptySet[A] =
+    NonEmptySet(head)
+
+  /** Creates a `NonEmptySet` containing elements from `l` and `r` */
+  def union[A](l: NonEmptySet[A], r: Set[A]): NonEmptySet[A] = {
+    val (head, tail) = l.destruct
+    NonEmptySet.fromSet(head, tail.union(r))
+  }
+
+  /** Creates a `NonEmptySet` containing elements from `l` and `r` */
+  def union[A](l: Set[A], r: NonEmptySet[A]): NonEmptySet[A] =
+    union(r, l)
+
+  /**
+   * The `Associative` instance for `NonEmptySet`.
+   */
+  implicit def NonEmptySetCommutative[A]: Commutative[NonEmptySet[A]] =
+    Commutative.make(_ ++ _)
+
+  /**
+   * Derives a `Debug[NonEmptySet[A]]` given a `Debug[A]`.
+   */
+  implicit def NonEmptySetDebug[A: Debug]: Debug[NonEmptySet[A]] =
+    chunk => Debug.Repr.VConstructor(List("zio", "prelude"), "NonEmptySet", chunk.toNonEmptyList.map(_.debug).toCons)
+
+  /**
+   * Derives an `Equal[NonEmptyList[A]]` given an `Equal[A]`.
+   */
+  implicit def NonEmptySetHash[A]: Hash[NonEmptySet[A]] =
+    Hash.default
+
+  /**
+   * Provides an implicit conversion from `NonEmptySet` to the `Set`
+   * for interoperability with Scala's collection library.
+   */
+  implicit def toSet[A](nonEmptySet: NonEmptySet[A]): Set[A] =
+    nonEmptySet.toSet
+
+}
+
+trait NonEmptySetSyntax {
+  implicit class SetOps[A](private val set: Set[A]) {
```

Comment:
Have you checked whether these implicit syntax methods work or get shadowed by the normal methods with the same name?

Reply:
they don't :disappointed: 
I've removed them

Reply:
If you want we could add a different name for them. Like I think we used concat for Chunk for that reason. Though it is implicit syntax on every Set and I don't know if it is the most important use case.

Reply:
I think we can try living without them and then we'll see